pgdog/src/backend/pool/cluster.rs

```diff
@@ -60,8 +63,26 @@ impl ShardingSchema {
     pub fn tables(&self) -> &ShardedTables {
         &self.tables
     }
+
+    pub fn has_global_sharding_key_with_mappings(&self) -> bool {
+        let mut global_sharding_rules_with_mappings = 0;
```

Comment:
Checkout the `map`, `filter`, `count`, and other iterator functions. Looks prettier!

```rust
let count = self.tables.tables().iter().filter(|table| table.name.is_none() && table.mapping.is_some()).count();

count == self.tables.tables.len()
```